server/proxy/bindings_test.go

```diff
@@ -461,3 +476,152 @@ func TestCleanAppBinding(t *testing.T) {
+				App:        apps.ExpandNone,
+				ActingUser: apps.ExpandNone,
+			},
+			Path: "/",
+		},
+	}
+
+	type TC struct {
+		name             string
+		applications     []apps.App
+		callRequest      apps.CallRequest
+		callResponse     apps.CallResponse
+		checkExpectation func(api *plugintest.API)
+	}
+
+	makeBindingRequest := func(req apps.CallRequest) apps.CallRequest {
+		req.Call.Path = path.Bindings
+		return req
+	}
+
+	for _, tc := range []TC{
+		{
+			name:         "refresh bindings when flag is set and OK response",
+			applications: tApps,
+			callRequest:  creq,
+			callResponse: apps.CallResponse{
+				Type:            apps.CallResponseTypeOK,
+				RefreshBindings: true,
+			},
+			checkExpectation: func(testApi *plugintest.API) {
+				testApi.On("PublishWebSocketEvent", config.WebSocketEventRefreshBindings, map[string]interface{}{}, &model.WebsocketBroadcast{UserId: "userid"}).Once()
+			},
+		},
+		{
+			name:         "don't refresh bindings when flag is set and Error response",
+			applications: tApps,
+			callRequest:  creq,
+			callResponse: apps.CallResponse{
+				Type:            apps.CallResponseTypeError,
+				RefreshBindings: true,
+			},
+			checkExpectation: func(testApi *plugintest.API) {
+			},
+		},
+		{
+			name:         "don't refresh bindings when flag is not set",
+			applications: tApps,
+			callRequest:  creq,
+			callResponse: apps.CallResponse{
+				Type:            apps.CallResponseTypeOK,
+				RefreshBindings: false,
+			},
+			checkExpectation: func(testApi *plugintest.API) {
+			},
+		},
+		{
+			name:         "don't refresh when binding response is handled",
+			applications: tApps,
+			callRequest:  makeBindingRequest(creq),
+			callResponse: apps.CallResponse{
+				Type:            apps.CallResponseTypeOK,
+				RefreshBindings: true,
+			},
+			checkExpectation: func(testApi *plugintest.API) {
+			},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			testAPI := &plugintest.API{}
+			testDriver := &plugintest.Driver{}
+			ctrl := gomock.NewController(t)
+
+			conf := config.NewTestConfigService(nil).WithMattermostConfig(model.Config{
+				ServiceSettings: model.ServiceSettings{
+					SiteURL: model.NewString("test.mattermost.com"),
+				},
+			}).WithMattermostAPI(pluginapi.NewClient(testAPI, testDriver))
+
+			s, err := store.MakeService(utils.NewTestLogger(), conf, nil)
+			require.NoError(t, err)
+			appStore := mock_store.NewMockAppStore(ctrl)
+			s.App = appStore
+
+			upstreams := map[apps.AppID]upstream.Upstream{}
+			for i := range tc.applications {
+				app := tc.applications[i]
+
+				up := mock_upstream.NewMockUpstream(ctrl)
+
+				// set up an empty OK call response
+				b, _ := json.Marshal(tc.callResponse)
+				reader := ioutil.NopCloser(bytes.NewReader(b))
+				up.EXPECT().Roundtrip(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(reader, nil)
+
+				upstreams[app.Manifest.AppID] = up
+				appStore.EXPECT().Get(app.Manifest.AppID).Return(&app, nil)
+			}
```

Comment:
Not a blocker. I would like to see an restapi test (`test/restapitest`) instead of all of these mocks. That way the test ensures that the websocket event is actually sent to the client.